crates/abq_workers/src/results_handler.rs

```diff
@@ -16,14 +16,14 @@ use crate::test_fetching;
 
 /// A results handler that dispatches results to the remote queue, and also any local handlers.
 pub(crate) struct MultiplexingResultsHandler {
-    remote_handler: QueueResultsSender,
+    remote_handler: Box<dyn NotifyResults + Send>,
```

Comment:
What do you think about making this `Option<QueueResultsHandler>` rather than an opaque interface object? I think it makes the intention that we either have results reported the queue, or we don't, more clear.

Reply:
Nice, yeah, I think that reads a lot nicer. Working through the types.